cmd/generate/generate.go

```diff
@@ -0,0 +1,285 @@
+package main
+
+import (
+	"embed"
+	"errors"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime/debug"
+	"strings"
+	"text/template"
+
+	"github.com/spf13/pflag"
+	"go.uber.org/zap/buffer"
+	"golang.org/x/mod/modfile"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift-knative/hack/pkg/project"
+	"github.com/openshift-knative/hack/pkg/prowgen"
+)
+
+const (
+	GenerateDockerfileOption = "dockerfile"
+)
+
+//go:embed Dockerfile.template
+var DockerfileTemplate embed.FS
+
+//go:embed BuildImageDockerfile.template
+var DockerfileBuildImageTemplate embed.FS
+
+func main() {
```

Comment:
very long, but not really a concern. Let's get this out, and improve later

Reply:
Yes, not the cleanest code